libs/libencrypt/CMakeLists.txt

```diff
@@ -0,0 +1,51 @@
+cmake_minimum_required(VERSION 3.16)
+project(libencrypt)
+
+find_package(PkgConfig REQUIRED)
+pkg_search_module(OPENSSL REQUIRED openssl)
+
+#file(READ "public_key.pem" PUBLIC_KEY_CONTENTS)
+file(READ "config.txt" PASSPHRASE_CONTENTS)
+
+#string(REPLACE "\n" "\\n" PUBLIC_KEY_CONTENTS "${PUBLIC_KEY_CONTENTS}")
+string(REPLACE "\n" "\\n" PASSPHRASE_CONTENTS "${PASSPHRASE_CONTENTS}")
+#set(PUBLIC_KEY "${PUBLIC_KEY_CONTENTS}")
+set(PASS_PHRASE "${PASSPHRASE_CONTENTS}")
+add_compile_definitions(PASS_PHRASE="${PASS_PHRASE}")
+
+if(OPENSSL_FOUND)
+    include_directories(${OPENSSL_INCLUDE_DIRS})
+    message(STATUS "Using OpenSSL ${OPENSSL_VERSION}")
+else()
+    message(FATAL_ERROR "OpenSSL not found")
+endif()
+
+# Source files for the library
+file(GLOB_RECURSE LIB_SOURCES src/*.cpp ../EXTERNAL/libpugixml/src/*.cpp)
+file(GLOB_RECURSE LIB_HEADERS src/*.hpp src/*.h  ../EXTERNAL/libpugixml/src/*.hpp)
```

Comment:
You donot need to glob these files here. You can simply add the ``libpugixml`` to the ``target_linked_libraries``

For example, 

https://github.com/verilog-to-routing/vtr-verilog-to-routing/blob/ec634df07a5be186a0e7b66f2789c4daf2e4b44c/libs/librrgraph/CMakeLists.txt#L24-L28